base_jsonify/README.rst

```diff
@@ -79,6 +79,116 @@ you can pass a callable or the name of a method on the model:
 Also the module provide a method "get_json_parser" on the ir.exports object
 that generate a parser from an ir.exports configuration.
 
+Further features are available for advanced uses.
+It defines a simple "resolver" model that has a "python_code" field and an eval
+function so that arbitrary functions can be configured to transform fields,
+or process the resulting dictionary.
+It is also to specify a lang to extract the translation of any given field.
+
+To use these features, a full parser follows the following structure:
+
+.. code-block:: python
+
+    parser = {
+        "resolver": 3,
+        "language_agnostic": True,
+        "langs": {
+            False: [
+                {'name': 'description'},
+                {'name': 'number', 'resolver': 5},
+                ({'name': 'partner_id', 'target': 'partner'}, [{'name': 'display_name'}])
+            ],
+            'fr_FR': [
+                {'name': 'description', 'target': 'descriptions_fr'},
+                ({'name': 'partner_id', 'target': 'partner'}, [{'name': 'description', 'target': 'description_fr'}])
+            ],
+        }
+    }
+
+
+One would get the a result having this structure (note that the translated fields are merged in the same dictionary):
+
+.. code-block:: python
+
+    exported_json == {
+        "description": "English description",
+        "description_fr": "French description, voilà",
+        "number": 42,
+        "partner": {
+            "display_name": "partner name",
+            "description_fr": "French description of that partner",
+        },
+    }
+
+
+Note that a resolver can be passed either as a recordset or as an id, so as to be fully serializable.
+A slightly simpler version in case the translation of fields is not needed,
+but other features like custom resolvers are:
+
+.. code-block:: python
+
+    parser = {
+        "resolver": 3,
+        "fields": [
+                {'name': 'description'},
+                {'name': 'number', 'resolver': 5},
+                ({'name': 'partner_id', 'target': 'partners'}, [{'name': 'display_name'}]),
+        ],
+    }
+
+
+By passing the `fields` key instead of `langs`, we have essentially the same behaviour as simple parsers,
+with the added benefit of being able to use resolvers.
+
+Standard use-cases of resolvers are:
+- give field-specific defaults (e.g. `""` instead of `None`)
+- cast a field type (e.g. `int()`)
+- alias a particular field for a specific export
+- ...
+
+A simple parser is simply translated into a full parser at export.
+
+If the global resolver is given, then the json_dict goes through:
+
+.. code-block:: python
+
+    resolver.eval(dict, record)
+
+Which allows to add external data from the context or transform the dictionary
+if necessary. Similarly if given for a field the resolver evaluates the result.
+
+It is possible for a target to end with a '*':
```

Comment:
can we make this more explicit?
See my example here https://github.com/OCA/website-cms/blob/11.0/cms_form/marshallers.py 
In that case if an input comes from the request and the name has a marshaller in it the value gets converted.
In your case I'd replace `*` with `:list`. This would open the door to support other marshallers in an explicit way as I did.